packages/utils/src/sortBlogPosts.ts

```diff
@@ -0,0 +1,34 @@
+import { compareAsc, compareDesc } from 'date-fns'
+import type { SortType } from './types/sortTypes'
+import type { WithPublishedOn } from './types/withPublishedOnType'
+
+export function sortPostsByDateDesc<Entry extends WithPublishedOn>(
+  posts: Array<Entry>,
+) {
+  return sortPostsByDate(posts, 'newest')
+}
+
+export function sortPostsByDateAsc<Entry extends WithPublishedOn>(
+  posts: Array<Entry>,
+) {
+  return sortPostsByDate(posts, 'oldest')
+}
+
+function sortPostsByDate<Entry extends WithPublishedOn>(
+  posts: Array<Entry>,
+  sortBy: SortType<'newest' | 'oldest'>,
```

Comment:
Could this just be `'newest' | 'oldest'`?

What does `SortType` add in this implementation?

```suggestion
  sortBy: 'newest' | 'oldest'
```